src/common/api/base.go

```diff
@@ -248,3 +249,47 @@ func (b *BaseAPI) SendPreconditionFailedError(err error) {
 func (b *BaseAPI) SendStatusServiceUnavailableError(err error) {
 	b.RenderFormattedError(http.StatusServiceUnavailable, err.Error())
 }
+
+// SendError return the error defined in OCI spec: https://github.com/opencontainers/distribution-spec/blob/master/spec.md#errors
+// {
+//	"errors:" [{
+//			"code": <error identifier>,
+//			"message": <message describing condition>,
+//			"detail": <unstructured>
+//		},
+//		...
+//	]
+// }
+func (b *BaseAPI) SendError(err error) {
+	var statusCode int
+	var send error
+
+	var e pkg_errors.Error
+	if errors.Is(err, e) {
```

Comment:
looks like we should use `errors.As` here.

Reply:
And I have doubts that in controller  how can I check the `error` is a `not found` error returned by dao.

Use `errors.As` and check the code of the error? This is very tricky.